src/main/java/org/apereo/openlrs/OpenLRSAuthenticationFilter.java

```diff
@@ -111,11 +111,24 @@ private void authenticateOAuth(String authorizationHeader, HttpServletRequest re
 						normalizedParams.put(key, value);
 					}
 				}
-				
+
+				StringBuffer requestURL = request.getRequestURL();
+				String requestScheme = request.getScheme();
+				String originalScheme = request.getHeader("x-forwarded-proto");
+
+				// If original request scheme (maybe through load-balancer) is different, use it instead.
```

Comment:
Good point.  I was wondering about that.  That gets a little trickier to implement, though.  The port isn't always included in the URL returned by `request.getRequestURL()`, like port 80 for HTTP, etc.

I want to avoid reimplementing `request.getRequestURL()`, because it could be wrong.  Since `HttpServletRequest` instances don't allow changing the scheme and port, I'm thinking of creating a new instance with the scheme and port we have and then calling its `request.getRequestURL()` to produce the new URL for calculating the signature.

If that's not possible, maybe subclassing `HttpServletRequest` with methods to allow setting the scheme and port.

Finally, if that's difficult, wrong, or kludgy, then fall back to searching for one port number and replace it with another.  It would also need to add the port in some cases.